tests/unit/test_column_noise.py

```diff
@@ -20,96 +31,183 @@
     pass
 
 
 @pytest.mark.skip(reason="TODO")
-def test_phonetic_noise():
+def test_generate_phonetic_errors():
     pass
 
 
 @pytest.mark.skip(reason="TODO")
-def test_ocr_noise():
+def test_generate_ocr_errors():
     pass
 
 
-@pytest.mark.skip(reason="TODO")
-def test_typographic_noise():
-    pass
+@pytest.mark.parametrize(
+    "column",
+    [
+        "numbers",
+        "characters",
+    ],
+)
+def test_generate_typographical_errors(dummy_dataset, column):
+    data = dummy_dataset[column]
+    config = get_configuration()
+    config.update(
+        {
+            "decennial_census": {
+                column: {
+                    "typographic": {
+                        "row_noise_level": 0.1,
+                        "token_noise_level": 0.1,
+                        "include_original_token_level": 0.1,
+                    },
+                },
+            },
+        }
+    )
+    config = config["decennial_census"][column]["typographic"]
+    noised_data = _validate_seed_and_noise_data(
+        func=generate_typographical_errors, column=data, config=config
+    )
+
+    not_missing_idx = data.index[(data.notna()) & (data != "")]
+    check_original = data.loc[not_missing_idx]
+    check_noised = noised_data.loc[not_missing_idx]
+
+    # Check for expected noise level
+    p_row_noise = config.row_noise_level
+    p_token_noise = config.token_noise_level
+    str_lengths = check_original.str.len()  # pd.Series
+    p_token_not_noised = 1 - p_token_noise
+    p_strings_not_noised = p_token_not_noised**str_lengths  # pd.Series
+    p_strings_noised = 1 - p_strings_not_noised  # pd.Series
+    expected_noise = p_row_noise * p_strings_noised.mean()
+    actual_noise = (check_noised != check_original).mean()
+    assert np.isclose(expected_noise, actual_noise, rtol=0.06)
+
+    # Check for expected string growth due to keeping original noised token
+    assert (check_noised.str.len() >= check_original.str.len()).all()
+    p_include_original_token = config.include_original_token_level
+    p_token_does_not_increase_string_length = 1 - p_token_noise * p_include_original_token
+    p_strings_do_not_increase_length = (
+        p_token_does_not_increase_string_length**str_lengths
+    )  # pd.Series
+    p_strings_increase_length = 1 - p_strings_do_not_increase_length  # pd.Series
+    expected_changed_length = p_row_noise * p_strings_increase_length.mean()
+    actual_changed_length = (check_noised.str.len() != check_original.str.len()).mean()
+    assert np.isclose(expected_changed_length, actual_changed_length, rtol=0.06)
+
+    # Check that we did not touch the missing data
+    assert (
+        data.loc[~data.index.isin(not_missing_idx)]
+        == noised_data.loc[~noised_data.index.isin(not_missing_idx)]
+    ).all()
+
+
+####################
+# HELPER FUNCTIONS #
+####################
+
+
+# TODO: refactor this into its own test parameterized by noise functions
+def _validate_seed_and_noise_data(func, column, config):
```

Comment:
A followup PR will move this method out of the existing tests and do these checks as their own tests.